src/libNetwork/P2PComm.cpp

```diff
@@ -344,23 +343,118 @@ void P2PComm::SendBroadcastMessageCore(
 void P2PComm::ClearBroadcastHashAsync(const vector<unsigned char>& message_hash)
 {
     LOG_MARKER();
-    lock_guard<mutex> guard(m_broadcastToRemovedMutex);
-    m_broadcastToRemoved.emplace_back(message_hash,
-                                      chrono::system_clock::now());
```

Comment:
The old way to detach a thread to track the message hash is quite thread-consuming. We've changed the code to use a buffer to store the hashes of broadcast messages and periodically clear the buffer.